beluga/test/beluga/sensor/test_likelihood_field_model_base.cpp

```diff
@@ -59,4 +59,53 @@ TEST(LikelihoodFieldModelBase, LikelihoodField) {
       testing::Pointwise(testing::DoubleNear(0.003), expected_likelihood_field));
 }
 
+TEST(LikelihoodFieldModelBase, PreProcessThickWalls) {
+  // Create a 5x5 grid with a 3x3 thick wall of obstacles in the center.
+  // The grid has a resolution of 1.0 m.
+  // F F F F F
+  // F O O O F
+  // F O O O F
+  // F O O O F
+  // F F F F F
+  const auto grid = beluga::testing::StaticOccupancyGrid<5, 5>{
+      {false, false, false, false, false,  //
+       false, true,  true,  true,  false,  //
+       false, true,  true,  true,  false,  //
+       false, true,  true,  true,  false,  //
+       false, false, false, false, false},
+      1.0};
+
+  auto params = beluga::LikelihoodFieldModelBaseParam{};
+  params.pre_process_thick_walls = true;
+  params.max_obstacle_distance = 10.0;
+  params.sigma_hit = 0.2;
+  params.z_hit = 0.5;
+  params.z_random = 0.5;
+  params.max_laser_distance = 2.0;
+
+  auto model = UUT{params, grid};
+
+  const auto& likelihood_field = model.likelihood_field();
+
+  // With pre_process_thick_walls=true, the center obstacle cell (12) at (2,2)
+  // should be treated as unknown, and its distance should be calculated
+  // from the boundary cells (e.g., cell 7 at (2,1)), which is 1.0 m.
+  // The boundary cells themselves should have a distance of 0.
+
+  const double two_squared_sigma = 2 * params.sigma_hit * params.sigma_hit;
+  const double amplitude = params.z_hit / (params.sigma_hit * std::sqrt(2 * Sophus::Constants<double>::pi()));
+  const double offset = params.z_random / params.max_laser_distance;
+
+  const auto to_likelihood = [=](double squared_distance) {
+    return amplitude * std::exp(-squared_distance / two_squared_sigma) + offset;
+  };
+
+  // Cell at (0,0) is free. Distance to nearest obstacle (cell 6 at (1,1)) is sqrt(2).
+  ASSERT_NEAR(likelihood_field.data_at(0, 0).value(), to_likelihood(2.0), 1e-6);
+  // Cell at (1,1) is a boundary obstacle. Distance should be 0.
+  ASSERT_NEAR(likelihood_field.data_at(1, 1).value(), to_likelihood(0.0), 1e-6);
+  // Cell at (2,2) is an inner obstacle. It should be treated as free space, and its distance to the boundary is 1.0.
```

Comment:
@LaBruma treated as free space or treated as not an obstacle? It probably doesn't change the test itself but the comment can be misleading.